### PR TITLE
EditableDatumLabel add support for angle labels.

### PR DESCRIPTION
**Pull request by PaddleStroke**
Following discussion on https://github.com/FreeCAD/FreeCAD/pull/10892
@abdullahtahiriyo cherry pick changes from editable datum label.

### CODE REVIEW

File: src/Gui/EditableDatumLabel.cpp [hunk 1]
```diff
@@ -171,15 +172,15 @@ double EditableDatumLabel::getValue()
     return spinBox->rawValue();
 }
 
-void EditableDatumLabel::setSpinboxValue(double val)
+void EditableDatumLabel::setSpinboxValue(double val, Base::Unit unit)
```
Comment (abdullahtahiriyo) [inline src/Gui/EditableDatumLabel.cpp:175]: Look at the lint just below.  It does not look like we need a copy of the unit, do we? If do not, consider ``` const Base::Unit & unit```. Sorry, I did not catch that one yesterday.

**Comment by PaddleStroke** [comment]
I have added the & and also a 'isInEdit' function.

**Comment by abdullahtahiriyo** [comment]
@PaddleStroke 

Why did you disregard the const when adding the reference?

As I see, the unit is only used as a parameter to the Quantity constructor, which takes a const &. It is not changed inside the function (and it would be difficult to think of an scenario where it does).

```explicit Quantity(double value, const Unit& unit=Unit()); ```

There is a good reason to declare const what won't change, it is called const correctness and shows the intent. There is, arguably, an additional reason (of lesser importance) related to enabling the compiler to do certain optimisations.

**Comment by PaddleStroke** [comment]
I have not disregarded it, in fact I have misread the snippet you provided sorry. I will update.

**Comment by abdullahtahiriyo** [comment]
No problem. 

I am more interested in that you understand why it should be const. That way you will  tend to apply it in the future.

**Pull request merged**